lib/docker.js

```diff
@@ -304,6 +304,11 @@ function dockerDownloadAndImportLayer(opts, callback) {
     }
 
     function uncompressAndSha256Contents(arg, next) {
+        // streampeek() inside of detectCompression leaves the stream unpaused
```

Comment:
One small question... why not put the stream.pause() in detectCompression()?  I can think of a few answers, but I'd like to hear yours, and have it on the record in the ticket. I didn't see an obvious why-here explanation save, via-experimentation (one of the few answers I thought of).

Reply:
Good question. I tried that. I inserted explicit pause on every line of detectCompression and it didn't work. I think it's because `streampeek` has an open `stream.on('data', ...)` reference that prevents the stream from being paused, but I really don't know a lot about streams so I'm not entirely sure here. This is arguably a bug in streampeek in that it should return the stream the way it found it (paused or not), and maybe it could be fixed inside of streampeek. There are several return jumps inside of `detectCompression`, so even if it did work by adding a pause before each of them, having a single that could always take care of it is better anyway.

As it turns out, putting that at the beginning of uncompressAndSha256Contents was the right place. It seems that the reason this did the trick was because starting to uncompressed the stream and pass it on is where we read from it before pausing it again. So as long as it's paused before we start decompressing it, we're golden.